File: src/main/scala/cub/hackerrank/tutorials/a30DaysOfCode/day25/day25.py
#!/usr/bin/env python3

# Enter your code here. Read input from STDIN. Print output to STDOUT

import math

T = int(input())
for i in range(T):
    data = int(input())
    isPrime = (data != 1 and data % 2 != 0) or data == 2
    j = 3
    while isPrime and j < math.sqrt(data) + 1:
        isPrime = data % j != 0
        j += 2
    print("Prime" if isPrime else "Not prime")

File: src/main/java/cub/hackerrank/tutorials/a30DaysOfCode/day25/day25.pl
#!/usr/bin/env perl

# Enter your code here. Read input from STDIN. Print output to STDOUT

$n = <STDIN>;
chomp $n;
for (1 .. $n) {
  my $data = <STDIN>;
  chomp $data;
  my $isPrime = ($data != 1 && $data % 2 != 0) || $data == 2;
  my $j = 3;
  while ($isPrime && $j < sqrt($data) + 1) {
    $isPrime = $data % $j != 0;
    $j += 2;
  }
  print($isPrime ? "Prime\n" : "Not prime\n");
}
